### DISCUSSION TITLE
CC1101 data rate tolerance in receive mode

### DISCUSSION THREAD

**Discussion by OlliLausS** [jgromes/RadioLib#1148] [Q&A] [answered]
I am currently struggling with CC1101 data reception at 868MHz. Data rates of some sensors I want to receive seem to have high tolerances and I was not able to figure out a data rate that fits all sensors I want to receive. In CC1101 datasheet a register named BSCFG is mentioned where configuration of data rate offset is possible. Radiolib standard configuration obviously is set to "no offset compensation". The CC1101 allows for max. 12,5% data rate offset when BSCFG is configured accordingly. I tried with a dirty hack and it works. Is there also a possibility in the lib to configure BSCFG register (or the data rate offset) with available functionality of the lib ?
Did not find anything in the docs.

**Comment by jgromes** [comment, marked as answer]
Currently, there is no public API to perform this operation on the CC1101. I didn't implement that because it is quite unique for CC1101 compared to the other supported radio modules.

But I do see the utility of that, so this could be added to the public API if you're willing to send a pull request. I think a simple `CC1101::setBitRateTolerance` method could do the trick.
